src/cosmos/CosmosEngine.ts

```diff
@@ -559,7 +557,10 @@ export class CosmosEngine extends CurrencyEngine<
           this.otherData[query]?.newestTxid !== newestTxid
         ) {
           this.otherData[query] = { newestTxid }
-          this.otherData.archivedTxLastCheckTime = lastTimestamp
+          this.otherData.archivedTxLastCheckTime = Math.max(
```

Comment:
approved with requested small change of only setting archivedTxLastCheckTime once both send and recv transactions sycned.